analyzers/tests/SonarAnalyzer.UnitTest/TestCases/ConditionalsWithSameCondition.CSharp9.cs

```diff
@@ -52,11 +52,25 @@ public void Foo()
                 f = cond ? new Apple() : new Orange();
             }
 
+            if (f is null) { }
+            if (f is null) { }  // Noncompliant
+
+            char ch = 'x';
+            if (ch is >= 'a' and <= 'z') { }
+            if (ch is >= 'a' and <= 'z') { } // Noncompliant
+            if (ch is <= 'z' and >= 'a') { } // Compliant, even when the semantics is the same
```

Comment:
could you also add some more complex patterns like 
```
c is (>= 'a' and <= 'z') or (>= 'A' and <= 'Z') or '.' or ',';
```
and
```
c is (>= 'A' and <= 'Z')  or (>= 'a' and <= 'z') or '.' or ',';
```
(I just swapped the first two)